olp-cpp-sdk-authentication/src/AuthenticationClientImpl.cpp

```diff
@@ -469,8 +501,14 @@ client::CancellationToken AuthenticationClientImpl::HandleUserRequest(
 
     client::OlpClient client = CreateOlpClient(settings_, boost::none, false);
 
-    std::time_t timestamp =
-        std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+    Timer timer;
+
+    if (!settings_.use_system_time) {
+      auto server_time = GetTimeFromServer(context, client);
```

Comment:
Is it not better than that GetTimeFromServer() returns directly a `Timer` instance, or whatever name it might have?